openfl/utilities/path_check.py

```diff
@@ -0,0 +1,17 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""openfl path checks."""
+
+import os
+from pathlib import Path
+from typing import Union
+
+
+def is_directory_traversal(directory: Union[str, Path]) -> bool:
+    """Check for directory traversal."""
+    cwd = os.path.abspath(os.getcwd())
+    requested_path = os.path.relpath(directory, start=cwd)
+    requested_path = os.path.abspath(requested_path)
+    common_prefix = os.path.commonprefix([requested_path, cwd])
+    return common_prefix != cwd
```

Comment:
```suggestion
    if common_prefix != cwd:
        raise RuntimeError(f"{requested_path} is out of the openfl workspace scope.")
```
Just to keep some information about what happened, if we rename it to `check_directory_traversal(some_path)`

Reply:
And then we get a generic exception into the user interface, if we don't catch it. It's expected error. It should be a human readable error description for humans, and error code for an operation system. Python exceptions are for python interpreters and developers.